### ISSUE TITLE
macOS: Menus pull-down onto the wrong screen

### ISSUE THREAD

**Issue by anjohnson** [eteran/nedit-ng#356]
I recently upgraded nedit-ng on my Apple M1 Macbook Pro to the commit c51fcc00476a4323f9a27fe65c404e569f04650e, at about the same time as updating it to macOS Sonoma 14.5, so I'm not sure exactly which change might have triggered this issue. I'm building against Qt 5.15.3, Bison 3.8.2 and Boost 1.78 installed with Homebrew which used to work fine for building an older nedit-ng, I haven't tried updating them yet.

With a single editor window open on my second monitor, when I click in the menu bar the menu gets displayed on the other (primary) monitor. The menu is shown correctly when the editor window is on the primary monitor.

Thanks!

**Comment by eteran** [comment]
Hmm, unfortunately, this sounds like a Qt bug. The reason why I say that is because there really isn't any code in nedit-ng that even tries to tell it where to place pull down menus (context menus like the right click is a different story). I'll poke around and see if I can find anything, but the answer may be "look into porting to Qt6"

**Comment by anjohnson** [comment]
Thanks, I will try updating my Qt5 installation, it's been a while since I did that and there is now a 5.15.13 release.

**Comment by anjohnson** [comment]
That fixed it, I'm now using Boost 1.85.0 and Qt 5.15.13.

FYI there were quite a few compiler warnings from Clang 15.0.0, but I think they were mostly repeats of this series:
```
/Users/anj/Software/other/nedit-ng/src/TextBufferFwd.h:8:44: warning: 'char_traits<unsigned char>' is deprecated: char_traits<T> for T not equal to char, wchar_t, char8_t, char16_t or char32_t is non-standard and is provided for a temporary period. It will be removed in LLVM 18, so please migrate off of it. [-Wdeprecated-declarations]
template <class Ch = char, class Tr = std::char_traits<Ch>>
                                           ^
/Users/anj/Software/other/nedit-ng/src/TextBufferFwd.h:12:21: note: in instantiation of default argument for 'BasicTextBuffer<unsigned char>' required here
using UTextBuffer = BasicTextBuffer<uint8_t>;
                    ^~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/usr/include/c++/v1/__string/char_traits.h:79:8: note: 'char_traits<unsigned char>' has been explicitly marked deprecated here
struct _LIBCPP_DEPRECATED_("char_traits<T> for T not equal to char, wchar_t, char8_t, char16_t or char32_t is non-standard and is provided for a temporary period. It will be removed in LLVM 18, so please migrate off of it.")
       ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/usr/include/c++/v1/__config:836:53: note: expanded from macro '_LIBCPP_DEPRECATED_'
#      define _LIBCPP_DEPRECATED_(m) __attribute__((deprecated(m)))
                                                    ^
```
